src/core/Akka.Tests/Dispatch/MailboxesSpec.cs

```diff
@@ -58,6 +59,68 @@ public IntPriorityMailbox(Settings settings, Config config) : base(settings, con
         }
     }
 
+    public class StashingActor : ReceiveActor, IWithUnboundedStash
+    {
+        private readonly TestKitBase _testkit;
+        private readonly bool _echoBackToSenderAsWell;
+        public IStash Stash { get; set; }
+
+        /// <summary>
+        /// TBD
+        /// </summary>
+        /// <param name="testkit">TBD</param>
+        /// <param name="echoBackToSenderAsWell">TBD</param>
+        public StashingActor(TestKitBase testkit, bool echoBackToSenderAsWell = true)
+        {
+            _testkit = testkit;
+            _echoBackToSenderAsWell = echoBackToSenderAsWell;
+            StashAll();
+        }
+
+        private void StashAll()
+        {
+            Receive<Start>(_ =>
+            {
+                Become(Process);
+                Stash.UnstashAll();
+            });
+            ReceiveAny(msg =>
+            {
+                Stash.Stash();
+            });
+        }
+
+        private void Process()
+        {
+            ReceiveAny(msg =>
+            {
+                var sender = Sender;
+                var testActor = _testkit.TestActor;
```

Comment:
This test actor was a quick copy paste from the EchoActor in the Testkit. But your right. It could be done simpler